### PR TITLE
braintree-web: paypal tokenize promise return type

### PR DESCRIPTION
**Pull request by tbrannam**
The promise returning version of `Paypal.tokenize()` returns a different type than the callback version.

Additionally, modified `callback` type to be support the Generic specification of expected type for instance.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
 - callback return signature https://github.com/braintree/braintree-web/blob/master/src/paypal/external/paypal.js#L297
 - promise return https://github.com/braintree/braintree-web/blob/master/src/paypal/external/paypal.js#L304

- [n/a] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [n/a] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## braintree-web ([<kbd>unpkg</kbd>](https://unpkg.com/browse/braintree-web@latest/))
was missing the following properties:
1. localPayment
1. masterpass
1. paymentRequest
1. vaultManager
1. visaCheckout

<details>
<summary>as well as these 1 other properties...</summary>
<p>preferredPaymentMethods</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ec3b91630dc88fa88a317c4606132a1890e9aa77
</p>

File: types/braintree-web/modules/paypal.d.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export interface PayPal {
      * });
      */
     create(options: { client: Client }): Promise<PayPal>;
-    create(options: { client: Client }, callback: callback): void;
+    create(options: { client: Client }, callback: callback<PayPal>): void;
```
Comment (jbuckner) [inline types/braintree-web/modules/paypal.d.ts:66]: Oh nice! So making the callback generic lets us specify the return type now?

**Review by jbuckner — APPROVED** [review]
Looks good! Thanks for the updates and tests!

**Comment by tbrannam** [comment]
Ready to merge

File: types/braintree-web/modules/paypal.d.ts (hunk 1 above)
Comment (tbrannam) [reply types/braintree-web/modules/paypal.d.ts:66]: Yup :)

**Pull request merged**